src/devices/MAX31856/MAX31856.cs

```diff
@@ -136,14 +133,15 @@ private Temperature ReadTemperature(out byte[] spiOutputData)
         /// <param name="spiOutputData">Spidata read from the device as 16 bytes</param>
         private double ConvertspiOutputDataTempColdJunction(byte[] spiOutputData)
         {
-            var tempRaw = (((spiOutputData[11] & 0x7F) << 8) + spiOutputData[12]);
-            double temperatureColdJunction = tempRaw / 256; // convert decimal ouput to temperature using a constant from the Technical Manual
+            Span<byte> reading = new Span<byte>(spiOutputData, 11, 2);
+            reading[0] = (byte)(spiOutputData[11] & 0x3F);
+            short tempRaw = BinaryPrimitives.ReadInt16BigEndian(reading);
             if ((spiOutputData[11] & 0x80) == 0x80) // checks if the temp is negative
             {
-                temperatureColdJunction = -temperatureColdJunction;
+                return -tempRaw / 256.0;
             }
 
-            return temperatureColdJunction;
+            return tempRaw / 256.0;
```

Comment:
ReadInt16BigEndian will do this part for you (you can just remove it), now you will double negate.

Reply:
disregard, per docs this looks correct